### PR TITLE
dedupe watchman recrawl warnings

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] needs review
- [x] needs test

### CODE REVIEW

File: src/watchman_watcher.js [hunk 1]
```diff
@@ -299,11 +299,54 @@ function handleError(self, error) {
  * @private
  */
 
-function handleWarning(resp) {
-  if ('warning' in resp) {
-    console.warn(resp.warning);
-    return true;
+var RECRAWL_WARNINGS = []; // shared structure, one per process.
+var REG = /Recrawled this watch (\d+) times, most recently because:\n([^:]+)/;
```
Comment (stefanpenner) [inline src/watchman_watcher.js]: this var name is bad, because of the 80char line length limit.

File: src/watchman_watcher.js [hunk 2]
```diff
@@ -299,11 +299,54 @@ function handleError(self, error) {
...
+      return false;
+    }
   } else {
+    RECRAWL_WARNINGS.push(new RecrawlWarning(root, count));
     return false;
   }
 }
+
+function handleWarning(resp) {
+	if ('warning' in resp) {
```
Comment (amasad) [inline src/watchman_watcher.js]: tabs?

File: src/watchman_watcher.js [hunk 3]
```diff
@@ -299,11 +299,54 @@ function handleError(self, error) {
...
+
+  if (warning) {
+    if (warning.count >= count ) {
+      return true;
+    } else {
+      warning.count = count;
+      return false;
+    }
   } else {
+    RECRAWL_WARNINGS.push(new RecrawlWarning(root, count));
```
Comment (amasad) [inline src/watchman_watcher.js]: nit: I'd just use a plain object, I don't see why it needs to be a class

**Comment by amasad** [comment]
looks good -- just nits

File: src/watchman_watcher.js (hunk 2 above)
Comment (stefanpenner) [inline src/watchman_watcher.js]: new laptop, will fix

**Comment by stefanpenner** [comment]
@amasad sorry got pulled into a meeting, anyways here are some updates + tests. 

Although I can merge, I can't release sane.

**Comment by stefanpenner** [comment]
@amasad got time to cut a release?

**Comment by amasad** [comment]
Oops sorry forgot to mention did that yesterday. Should be the latest

**Pull request merged**